test/topic_event.js

```diff
@@ -154,6 +148,81 @@ contract('TopicEvent', function(accounts) {
                 assert.match(e.message, regexInvalidOpcode);
             }
         });
+
+        it('throws if name is empty', async function() {
+            try {
+                await TopicEvent.new(testTopicParams._owner, testTopicParams._oracle, [], testTopicParams._resultNames, 
+                    testTopicParams._bettingEndBlock, testTopicParams._arbitrationOptionEndBlock);
+                assert.fail();
+            } catch(e) {
+                assert.match(e.message, regexInvalidOpcode);
+            }
+        });
+
+        it('throws if resultNames 0 or 1 are empty', async function() {
```

Comment:
Requires the first and second results non-empty. This condition feels a little hacky. Is there a way to do require( _resultNames.length >= 2), because basically we are restricting that resultsNames has at least two entries?

Reply:
On a second thought, restrictive rules like this should be fine.

Reply:
In Solidity, if we pass in an fixed array length, we are always passing in the max size of the array. So checking the length does not work here as it will always be 10. We are forced to check the data of the array values itself for its default 0 value.